### PR TITLE
QtSixa

### PR DESCRIPTION
**Pull request by maggu2810**
Send pull request as proposed here https://bugs.gentoo.org/show_bug.cgi?id=355497#c7

### CODE REVIEW

File: games-util/qtsixa/qtsixa-1.5.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Copyright 1999-2012 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Header: $
+
+EAPI=4
```
Comment (hasufell) [inline games-util/qtsixa/qtsixa-1.5.1.ebuild]: Use latest EAPI whenever possible.

File: games-util/qtsixa/qtsixa-1.5.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# Copyright 1999-2012 Gentoo Foundation
```
Comment (hasufell) [inline games-util/qtsixa/qtsixa-1.5.1.ebuild]: we have 2013

File: games-util/qtsixa/qtsixa-1.5.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+EAPI=4
+inherit eutils systemd
+
+MY_P="QtSixA-${PV}"
+S="${WORKDIR}/${MY_P}"
+
+DESCRIPTION="Sixaxis Joystick Manager"
+HOMEPAGE="http://qtsixa.sourceforge.net/"
+SRC_URI="http://sourceforge.net/projects/qtsixa/files/${MY_P/-/%20}/${MY_P}-src.tar.gz"
```
Comment (hasufell) [inline games-util/qtsixa/qtsixa-1.5.1.ebuild]: sourceforge is in thirdparymirrors

mirror://sourceforge/foo

File: games-util/qtsixa/qtsixa-1.5.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+LICENSE="GPL-2"
+
+SLOT="0"
+KEYWORDS="~x86 ~amd64"
```
Comment (hasufell) [inline games-util/qtsixa/qtsixa-1.5.1.ebuild]: order keywords alphabetically

File: games-util/qtsixa/qtsixa-1.5.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+IUSE="jack qt4"
+
+DEPEND="virtual/libusb:1
+	net-wireless/bluez
+	jack? ( media-sound/jack-audio-connection-kit )
+	qt4? ( dev-python/PyQt4 )"
+
```
Comment (hasufell) [inline games-util/qtsixa/qtsixa-1.5.1.ebuild:24]: order dependencies alphabetically too

useflag dependent deps go last and ordered by useflag name

File: games-util/qtsixa/qtsixa-1.5.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+	emake -C sixad
+}
+
+src_install() {
+	use qt4 && emake -C qtsixa install DESTDIR="${D}"
+	emake -C utils install DESTDIR="${D}" WANT_JACK=$(use jack && echo true)
+	emake -C sixad install DESTDIR="${D}"
+
+	dodoc INSTALL manual.pdf README TODO
+	rm "${D}etc/init.d/sixad" || die # TODO: Write a Gentoo version.
```
Comment (hasufell) [inline games-util/qtsixa/qtsixa-1.5.1.ebuild:59]: no openrc script?

File: games-util/qtsixa/qtsixa-1.5.1.ebuild [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+LICENSE="GPL-2"
+
+SLOT="0"
+KEYWORDS="~x86 ~amd64"
+IUSE="jack qt4"
+
+DEPEND="virtual/libusb:1
+	net-wireless/bluez
+	jack? ( media-sound/jack-audio-connection-kit )
+	qt4? ( dev-python/PyQt4 )"
```
Comment (hasufell) [inline games-util/qtsixa/qtsixa-1.5.1.ebuild:23]: this package depends on PyQt4 but does not use any python eclass

do we have any modules, scripts etc? make a gist of the file list and the build log and post it on the discussion page

File: games-util/qtsixa/qtsixa-1.5.1.ebuild (hunk 1 above)
Comment (maggu2810) [inline games-util/qtsixa/qtsixa-1.5.1.ebuild]: Done locally.

File: games-util/qtsixa/qtsixa-1.5.1.ebuild (hunk 2 above)
Comment (maggu2810) [inline games-util/qtsixa/qtsixa-1.5.1.ebuild]: Done locally.

File: games-util/qtsixa/qtsixa-1.5.1.ebuild (hunk 3 above)
Comment (maggu2810) [inline games-util/qtsixa/qtsixa-1.5.1.ebuild]: Done locally.

File: games-util/qtsixa/qtsixa-1.5.1.ebuild (hunk 4 above)
Comment (maggu2810) [inline games-util/qtsixa/qtsixa-1.5.1.ebuild]: Done locally.

File: games-util/qtsixa/qtsixa-1.5.1.ebuild (hunk 5 above)
Comment (maggu2810) [inline games-util/qtsixa/qtsixa-1.5.1.ebuild:24]: Done locally.

File: games-util/qtsixa/qtsixa-1.5.1.ebuild (hunk 6 above)
Comment (maggu2810) [inline games-util/qtsixa/qtsixa-1.5.1.ebuild:59]: There is currently no upstream one.

Comment (hasufell) [inline games-util/qtsixa/qtsixa-1.5.1.ebuild:59]: Then write one. I am a systemd hater, so I want an openrc script.

Comment (maggu2810) [inline games-util/qtsixa/qtsixa-1.5.1.ebuild:59]: All my gentoo machines running without openrc. So I will keep the ebuild in my overlay and cancel the pull request.
Thank you for your other tips. I will keep them in mind.

Comment (hasufell) [inline games-util/qtsixa/qtsixa-1.5.1.ebuild:59]: Write one, paste it, and I will test it.

http://www.gentoo.org/doc/en/handbook/handbook-x86.xml?part=2&chap=4

File: games-util/qtsixa/qtsixa-1.5.1.ebuild (hunk 7 above)
Comment (maggu2810) [inline games-util/qtsixa/qtsixa-1.5.1.ebuild:23]: Something done, have to be checked.

**Comment by maggu2810** [comment]
Sorry, did not get a email notification of your last comment (have to check why).
So, I will keep the working on, but no time ATM.

**Comment by maggu2810** [comment]
Sorry! The commits e2acec9, d337720 are not part of this pull request.

With 6065178 i fixed missing includes for recent toolchain and the two unconverted shebang.
Do not know how to fix the hardcoded python the right way (sed [replace with what], eclass function, ...).

Will add the missing dependencies later, but if you can show me how to support multiple python APIs, we have IMHO to use python-r1 instead of python-single-r1 and the dependencies have to be correct, too.

**Comment by hasufell** [comment]
No, python-r1 is not for single-ABI ebuilds.
And yes, you will sed it which is a poor fix, but works just fine for python-single-r1 ebuilds.
sed -i -e 's/python/${EPYTHON}/' ...

Also, generally patches should always be named this way: ${P}-topic.patch

If you have a bitbucket account please use that one, because github is giving me troubles and support does not give a f*ck: https://bitbucket.org/hasufell/sunrise

**Comment by maggu2810** [comment]
"Also, generally patches should always be named this way: ${P}-topic.patch"
It is okay for me, I will do it so. But then a lot of wrong stuff is in the official portage tree.

Sorry, I do not understand, why you wrote "No, python-r1 is not for single-ABI ebuilds".
I have written, the python-r1 have to be used (instead python-single-r1), if we would like to support multiple pyhton API. What statement did you give a "no"?

**Comment by hasufell** [comment]
multiple ABI is not trivial, so we should go with python-single-r1

it is very well possible that there is wrong stuff in the tree. Not setting $PV or $P in patch names will make problems on version bumps

**Comment by maggu2810** [comment]
Using my bitbucket account to create a new pull request with a more recent package.
Could you close / cancel the request here now?

**Pull request closed without merging**